Source/WebKit/GPUProcess/graphics/RemoteImageBufferSet.cpp

```diff
@@ -123,8 +123,10 @@ void RemoteImageBufferSet::ensureBufferForDisplay(const ImageBufferSetPrepareBuf
     bool needsFullDisplay = false;
     if (m_frontBuffer) {
         auto previousState = m_frontBuffer->setNonVolatile();
-        if (previousState == SetNonVolatileResult::Empty)
+        if (previousState == SetNonVolatileResult::Empty) {
             needsFullDisplay = true;
+            inputData.dirtyRegion = IntRect { { }, expandedIntSize(m_logicalSize) };
+        }
     }
 
     if (!m_frontBuffer || !inputData.supportsPartialRepaint || isSmallLayerBacking(m_frontBuffer->parameters()))
```

Comment:
also seems like these also imply a need for a layer-sized dirty region? maybe it happens elsewhere?

Reply:
It does! These are all things that the web process knew about in advance and has accounted for. Unfortunately we need to compute and check these things on both sides :(

Adding an assert that the dirtyRegion is already expanded appropriately would be good at least.